### PR TITLE
Fix time based backlog quota.

### LINKED ISSUE #11404: [Broker] Backlog Quota not properly fetched for message_age BacklogQuotaType
**Describe the bug**
pr https://github.com/apache/pulsar/pull/10093 introduced time based backlog and new BacklogQuotaType `message_age`.
While we can use admin cli to set backlog limit for `message_age` type, when fetching backlog https://github.com/apache/pulsar/blob/5ad405988fabb4b28dbdbd5aa5c9a10802f39af1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java#L73-L91, by default we only fetch for `destination_storage` type, so we set backlog to `message_age` type but when checking backlog we're not checking against limit we set for time based quota.

### PR DESCRIPTION
**Pull request by MarvinCai**
Fixes #11404

### Motivation
Time based backlog quota type message_age is set separately but when check backlog we're only checking against destination_storage type.
So fix to loop through all BacklogQuotaType when checking if backlog exceeded.

### Verifying this change
This change added tests and can be verified as follows:
  - *Added unit test.*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

Added default implementation to make Admin Topic/Namespace backlog quota related API backward compatible.

#### For contributor

For this PR, do we need to update docs? No

- Doc should be auto generated.

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
very good!

LGTM

probably we will have to update some docs (I am not sure)

**Comment by MarvinCai** [comment]
@Anonymitaet updated doc, please take a look.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java [hunk 1]
```diff
@@ -37,7 +38,7 @@ public static BacklogQuotaImplBuilder builder() {
     }
 
     public static class BacklogQuotaImplBuilder implements BacklogQuota.Builder {
-        private long limitSize;
+        private long limitSize = -1 * 1024 * 1024 * 1024;
```
Comment (codelipenghui) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java:41]: Why we need `-1 * 1024 * 1024 * 1024` not `-1`?

**Review by codelipenghui — APPROVED** [review]
LGTM, just left some minor comments.

**Review by Anonymitaet — APPROVED** [review]

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java (hunk 1 above)
Comment (hangc0276) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java:41]: the default backlog limit size is -1GB, we use `-1 * GB` and define `GB = 1024 * 1024 * 1024` is better.

**Review by hangc0276 — APPROVED** [review]
Good job!
I left a comment.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java (hunk 1 above)
Comment (MarvinCai) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/impl/BacklogQuotaImpl.java:41]: actually as long as it's negative, it doesn't really matter, but just want to keep it consistent with existing default value: https://github.com/apache/pulsar/blob/27dcb0a6febd0d88438e18fae72aad8ee38c0738/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java#L57

**Pull request merged**